pkg/transform/name_resolver.go

```diff
@@ -177,7 +177,12 @@ func (nr *NameResolver) resolveNames(span *request.Span) {
 	}
 
 	if span.IsClientSpan() {
-		hn, span.OtherNamespace = nr.resolve(&span.Service, span.Host)
+		hn = span.HostName
```

Comment:
This is exactly the same as before @marctc. The nr.resolve code will be returning an IP address (e.g. Host) only if we didn't find any host name. There fore we check that condision on line 186 below. Namely, if the returned resolved value is empty or it's the same as Host we try from the host scheme.

What you really want to do is if, the same condition is met, e.g. hn == "" || hn == span.Host, is to set hn to the prior host name if it's not empty.

An alternative is to pass a third argument to nr.resolve that is the previous value. So if dnsResolve didn't return a value, then we should return the existing value for the name if it was set.